src/loaders/ObjectLoader.js

```diff
@@ -455,6 +455,19 @@ Object.assign( THREE.ObjectLoader.prototype, {
 				case 'Scene':
 
 					object = new THREE.Scene();
+					
+					if ( data.fog !== undefined ) {
+						
+						if ( data.fog.density !== undefined ) {
```

Comment:
I would do instead `data.fog.type === 'FogExp2'` and add a `type` in the `toJSON()`.